lib/github.js

```diff
@@ -78,99 +107,214 @@ async function getOctokit (opts = {}) {
+    if (nextPage === true) {
+      arrayOfPRs.push.apply(arrayOfPRs, await getRemainingPullRequests(graphQL, owner, repo, cursor))
+    }
+    return arrayOfPRs
+  } catch (error) {
+    error.code = 'GET_REPO_PRS_FAILED'
+    Error.captureStackTrace(error, module.exports.getRepoIssues)
+    throw error
+  }
+}
+
+async function getPullRequests (graphQL, owner, repo) {
+  try {
+    const resp = await graphQL({
+      query: `query ($org: String!, $repo: String!) {
+        organization(login: $org) {
+          repository(name: $repo) {
+            pullRequests(first: 100, orderBy: {field: CREATED_AT, direction: DESC}) {
+              ${issueQuerySnip}
+            }
+          }
+        }
+      }`,
+      org: owner,
+      repo: repo
+    })
+    let arrayOfPRs = resp.organization.repository.pullRequests.edges
+    let cursor = resp.organization.repository.pullRequests.pageInfo.endCursor
+    let nextPage = resp.organization.repository.pullRequests.pageInfo.hasNextPage
+    if (nextPage === true) {
+      arrayOfPRs.push.apply(arrayOfPRs, await getRemainingPullRequests(graphQL, owner, repo, cursor))
     }
+    return arrayOfPRs
+  } catch (error) {
+    error.code = 'GET_REPO_PRS_FAILED'
+    Error.captureStackTrace(error, module.exports.getRepoIssues)
+    throw error
+  }
+}
 
-    for (let i of resp.data) {
-      yield new Issue(owner, repo, i)
+async function getRemainingIssues (graphQL, owner, repo, cursor) {
+  try {
+    const resp = await graphQL({
+      query: `query ($org: String!, $repo: String!, $cursor: String!) {
+        organization(login: $org) {
+          repository(name: $repo) {
+            issues(first: 100, after: $cursor, orderBy: {field: CREATED_AT, direction: DESC}) {
+              ${issueQuerySnip}
+            }
+          }
+        }
+      }
+      `,
+      org: owner,
+      repo: repo,
+      cursor: cursor
+    })
+    let arrayOfIssues = resp.organization.repository.issues.edges
+    const nextPage = resp.organization.repository.issues.pageInfo.hasNextPage
+    cursor = resp.organization.repository.issues.pageInfo.endCursor
+    if (nextPage === true) {
+      arrayOfIssues.push.apply(arrayOfIssues, await getRemainingIssues(graphQL, owner, repo, cursor))
     }
+    return arrayOfIssues
+  } catch (error) {
+    error.code = 'GET_REPO_ISSUES_FAILED'
+    Error.captureStackTrace(error, module.exports.getRepoIssues)
+    throw error
   }
 }
 
+module.exports.getRepoIssues =
+  async function * getRepoIssues (graphQL, owner, repo) {
+    try {
+      let resp = await graphQL({
+        query: `query ($org: String!, $repo: String!) {
+          organization(login: $org) {
+            repository(name: $repo) {
+              issues(first: 100, orderBy: {field: CREATED_AT, direction: DESC}) {
+                ${issueQuerySnip}
+              }
+            }
+          }
+        }`,
+        org: owner,
+        repo: repo
+      })
+      let arrayOfIssues = resp.organization.repository.issues.edges
```

Comment:
I think we can also simplify the push methodology you are using.  For example:

```

const { pageInfo, edges } = resp.organization.repository.issues
const issues = !pageInfo.nextPage ? edges : edges.concat(await getReminingIssues(graphQL, owner, repo, pageInfo.cursor))
```

Reply:
Thanks for the suggestion, updated the PR